src/modules/navigation/NavigationState.js

```diff
@@ -14,70 +17,83 @@ export function switchTab(index) {
 }
 
 // Action creators
-export function pushRoute(state) {
-  return (dispatch, getState) => {
-    // conditionally execute push to avoid double
-    // navigations due to impatient users
-    if (!isNavigationAnimationInProgress(getState())) {
-      dispatch({type: PUSH_ROUTE, payload: state});
```

Comment:
this condition was preventing any double pushed route, whereas it is better to prevent only double pushed route where the route is the same